src/libraries/System.Net.Http.Json/tests/FunctionalTests/JsonContentTests.cs

```diff
@@ -0,0 +1,212 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Http.Headers;
+using System.Net.Test.Common;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Json.Functional.Tests
+{
+    public class JsonContentTests
+    {
+
+        private class Foo { }
+        private class Bar { }
+
+        [Fact]
+        public void JsonContentObjectType()
+        {
+            Type fooType = typeof(Foo);
+            Foo foo = new Foo();
+
+            JsonContent content = JsonContent.Create(foo, fooType);
+            Assert.Equal(fooType, content.ObjectType);
+            Assert.Same(foo, content.Value);
+
+            content = JsonContent.Create(foo);
+            Assert.Equal(fooType, content.ObjectType);
+            Assert.Same(foo, content.Value);
+
+            object fooBoxed = foo;
+
+            // ObjectType is the specified type when using the .ctor.
+            content = JsonContent.Create(fooBoxed, fooType);
+            Assert.Equal(fooType, content.ObjectType);
+            Assert.Same(fooBoxed, content.Value);
+
+            // ObjectType is the declared type when using the factory method.
+            content = JsonContent.Create(fooBoxed);
+            Assert.Equal(typeof(object), content.ObjectType);
+            Assert.Same(fooBoxed, content.Value);
+        }
+
+        [Fact]
+        public void TestJsonContentMediaType()
+        {
+            Type fooType = typeof(Foo);
+            Foo foo = new Foo();
+
+            // Use the default content-type if none is provided.
+            JsonContent content = JsonContent.Create(foo, fooType);
+            Assert.Equal("application/json", content.Headers.ContentType.MediaType);
+            Assert.Equal("utf-8", content.Headers.ContentType.CharSet);
+
+            content = JsonContent.Create(foo);
+            Assert.Equal("application/json", content.Headers.ContentType.MediaType);
+            Assert.Equal("utf-8", content.Headers.ContentType.CharSet);
+
+            // Use the specified MediaTypeHeaderValue if provided.
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse("foo/bar; charset=utf-8");
+            content = JsonContent.Create(foo, fooType, mediaType);
+            Assert.Same(mediaType, content.Headers.ContentType);
+
+            content = JsonContent.Create(foo, mediaType: mediaType);
+            Assert.Same(mediaType, content.Headers.ContentType);
+        }
+
+        [Fact]
+        public async Task SendQuotedCharsetAsync()
+        {
+            JsonContent content = JsonContent.Create<Foo>(null);
+            content.Headers.ContentType.CharSet = "\"utf-8\"";
+
+            HttpClient client = new HttpClient();
+            var request = new HttpRequestMessage(HttpMethod.Post, "http://example.com");
+            request.Content = content;
+            await client.SendAsync(request);
```

Comment:
This test is actually making network calls out to http://example.com?  Is that intentional?

If we actually need to, we should mark the test as [Outerloop].  But I question whether that's necessary.

Reply:
Not intentionally, I just wanted to avoid writing the `LoopbackServer` code, since the only thing I care is to test the underlying call to `JsonSerializer.Serialize` before sending the request.

Reply:
> Using OuterLoop is not necessary either.

Meaning you're going to rewrite it to use the loopback server?

Reply:
Yes

Reply:
Ok, thanks.